src/screens/ConnectScreen.js

```diff
@@ -1,17 +1,249 @@
+    };
+  }
+
+  onRegionChange(region) {
+    this.setState({ region: region });
+  }
+
+  /**
+   * Called when a marker is pressed and does two things:
+   *   1. Animates map to center on coordinate
+   *   2. Opens appropriate ConnectBox or refreshes current active box
+   *
+   * @param {object} connection: either a veteran or parter org object
+   *                 with required fields as described in `ConnectBox`
+   * @return {() => null}: a function that animates the mapview and also
+   *                       opens the ConnectBox for this veteran/org
+   */
+  onMarkerPress(connection) {
+    const coordinate = {
+      latitude: connection.lat,
+      longitude: connection.lng,
+    };
+    const mapAnimateDuration = 100;
+    const connectBoxAnimateDuration = 200;
+
+    return () => {
+      // First need to animate map to current coordinate
+      this.mapView && this.mapView.animateToCoordinate(coordinate, mapAnimateDuration);
+
+      // Next also need to show Connect Box for this connection
+      this.openConnectBox(connection);
+    };
+  }
+
+  /**
+   * Renders a help modal with some information when the user first
+   * sees the page.
+   */
+  renderHelpModal() {
+    if (this.state.isHelpModalOpen) {
+      const { params } = this.props.navigation.state;
+      return (
+        <InfoModal
+          title={`WELCOME TO CONNECT, ${params.first_name}!`}
+          text="Pan the map to see who's around you."
+          onClose={this.closeHelpModal}
+        />
+      );
+    }
+  }
+
+  /**
+   * Return a list of MapView Markers that represent veterans.
+   * TODO (Ken): See if you can use underscore.js partials instead of
+   * rocket function to bind the onPress method
+   */
+  renderVeteranMarkers() {
+    return this.state.veterans.map((veteran) => {
+      const coordinate = {
+        latitude: veteran.lat,
+        longitude: veteran.lng,
+      };
+      return (
+        <MapView.Marker
+          coordinate={coordinate}
+          onPress={this.onMarkerPress(veteran)}
+          key={`veteran-${veteran.id}`}
+        >
+          <ConnectPin pinType="veteran" />
+        </MapView.Marker>
+      );
+    });
+  }
+
+  /**
+   * Return a list of MapView Markers that represent partering orgs.
+   * TODO (Ken): See if you can use underscore.js partials instead of
+   * rocket function to bind the onPress method
+   */
+  renderParterOrgMarkers() {
+    return this.state.parterOrgs.map((org) => {
+      const coordinate = {
+        latitude: org.lat,
+        longitude: org.lng,
+      };
+      return (
+        <MapView.Marker
+          coordinate={coordinate}
+          onPress={this.onMarkerPress(org)}
+          key={`parterOrg-${org.id}`}
+        >
+          <ConnectPin pinType="parterOrg" />
+        </MapView.Marker>
+      );
+    })
+  }
+
+  renderConnectBox() {
+    return this.state.activeConnection ? (
+      <ConnectBox
+        connection={this.state.activeConnection}
```

Comment:
is activeConnection also an object (PO or veteran) which is the veteran/org that has currently been focused on?

Reply:
yup its an object with all the fields and values